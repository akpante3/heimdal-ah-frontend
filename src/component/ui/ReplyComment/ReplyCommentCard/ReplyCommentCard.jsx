import React from 'react';
import FontAwesome from 'react-fontawesome';
import { connect } from 'react-redux';
import DeleteButton from '../../CommentForm/CommentDelete';
import './ReplyCommentCard.scss';

const ReplyCommentCard = (props) => {
  const { user } = props.auth;
  let replies = props.replies[props.commentId];
  replies = replies || [];
  const renderCard = () => replies.map(item => (
    <section key={item.id} className="comment-section container">
      <div className="comment-card row">
        <div className="comment-image col-md-2 col-sm-2">
          <img className="comment-card-img" src={item.profile.image} alt="user-comment-logo" />
        </div>
        <div className="comment-card-body col-md-9 col-sm-9">
          <div className="comment-body-top">
            <span className="reply-name">{item.profile.username}</span>
            <span className="comment-date small_text">
              <FontAwesome name="clock" className="fav_icons" />
            </span>
            <span className="small_text">
              <FontAwesome name="calendar" className="fav_icons" />
                09: 00 am
            </span>
            { item.userId === user.userId
              ? <DeleteButton replyId={item.id} buttonName='reply' commentId={props.commentId} />
              : <div />
            }
          </div>
          <div className="reply-body">{item.content}</div>
          <div className="comment-body-footer">
            <span />
            <span />
          </div>
        </div>
      </div>
    </section>
  ));

  return <div>{replies || Object.keys(replies).length !== 0 ? renderCard() : <div />}</div>;
};

const mapStateToProps = state => ({
  auth: state.auth,
  replies: state.replies,
});

export default connect(
  mapStateToProps,
  {},
)(ReplyCommentCard);
