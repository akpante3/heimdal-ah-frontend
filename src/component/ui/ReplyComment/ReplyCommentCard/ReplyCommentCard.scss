.reply-body {
  margin-left: 30px;
}
.reply-name {
  margin-left: 40px;
}

.small_text {
  margin-left: 20px;
}