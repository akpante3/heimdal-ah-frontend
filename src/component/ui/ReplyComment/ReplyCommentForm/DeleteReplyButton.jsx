import React from 'react';
import FontAwesome from 'react-fontawesome';
import { connect } from 'react-redux';


const DeleteButton = (props) => {
  const deleteComment = (e) => {
    e.preventDefault();
    const { commentId } = props;
    props.deleteArticleComment(1, commentId);
  };
  return (
    <button className='archive_button btn btn-danger' onClick={e => deleteComment(e)} type='submit'><FontAwesome name='trash' className='' /></button>
  );
};

const mapStateToProps = state => ({
  comment: state.comment,
});

export default connect(mapStateToProps, { })(DeleteButton);
