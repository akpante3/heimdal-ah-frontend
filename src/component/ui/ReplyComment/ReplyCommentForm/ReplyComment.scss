
.reply_button {
  cursor: pointer;
}
.reply_form {
  margin-top: 20px;
}

.reply_input {
  display: flex;
  justify-content: flex-end;
}

.btn__reply {
  float: right;
}

.error_msg {
  margin-top: 10px;
  text-align: end;
  margin-left: 50px;
}