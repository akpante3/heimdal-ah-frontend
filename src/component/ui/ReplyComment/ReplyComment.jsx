/* eslint-disable react/no-unused-prop-types */
/* eslint-disable react/forbid-prop-types */
/* eslint-disable react/require-default-props */
/* eslint-disable import/no-named-as-default */
/* eslint-disable no-unused-expressions */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import FontAwesome from 'react-fontawesome';
import ReplyCommentCard from './ReplyCommentCard/ReplyCommentCard';
import ReplyCommentForm from './ReplyCommentForm/ReplyCommentForm';
import { getCommentReplies } from '../../../actions/ReplyActions/ReplyAction';

/**
  * renderComponent
  * @method Class Class based Component
  * @summary React component for rendering the card components
  * @param {object}  replies contains the replies of a comment
  * @param {Object} props - React PropTypes
  * @property {String|Object} className - String className compatible object for styling
  * @return {Node} React node containing comment form view
  */
export class ReplyComment extends Component {
  state = {
    showReplies: false,
  };

  fetchReplies = (e) => {
    e.preventDefault();

    const { showReplies } = this.state;
    this.setState({
      showReplies: !showReplies,
    });
    !this.state.showReplies && this.props.getCommentReplies(this.props.commentId);
  };

  /**
     * Renders the component.
     *
     * @memberof app.components.ReplyComment
     * @return {string} - HTML markup for the component
   */
  render() {
    return (
      <div>
        <div
          role="searchbox"
          onKeyPress={e => this.fetchReplies(e)}
          onClick={e => this.fetchReplies(e)}
          className="reply_button"
          tabIndex="-1"
        >
          <FontAwesome name="comments" className="fav_icons" />
          Reply
        </div>
        {this.state.showReplies ? (
          <div>
            {this.props.replies ? (
              <div>
                {
                  this.props.commentId
                  && (
                  <div>
                    <ReplyCommentCard
                      replies={this.props.replies}
                      commentId={this.props.commentId}
                    />
                    <ReplyCommentForm commentId={this.props.commentId} />
                  </div>
                  )

                }
              </div>
            ) : (
              <div>No reply found</div>
            )}
          </div>
        ) : (
          <div />
        )}
      </div>
    );
  }
}

ReplyComment.propTypes = {
  getCommentReplies: PropTypes.func.isRequired,
  auth: PropTypes.object.isRequired,
  replies: PropTypes.object,
};

/**
 * @method module:Reactator.ReduxContainerBuilderMapStateToProps
 *
 * @param {function} mapStateToProps - function mapping redux state to props
 *
 * @return {ReduxContainerBuilder} this builder
*/
const mapStateToProps = state => ({
  auth: state.auth,
  replies: state.replies,
  replyFormError: state.replies.replyError,
  status: state.replies.status,
});

export default connect(
  mapStateToProps,
  { getCommentReplies },
)(ReplyComment);
