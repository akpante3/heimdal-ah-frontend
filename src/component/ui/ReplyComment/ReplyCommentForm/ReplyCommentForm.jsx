/* eslint-disable react/forbid-prop-types */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { FormGroup, FormControl } from 'react-bootstrap';
import { connect } from 'react-redux';
import './ReplyComment.scss';
import Button from '../../Buttons/Button';
import Alert from '../../Alert/Alert';
import { postCommentReplies } from '../../../../actions/ReplyActions/ReplyAction';


/**
  * renderComponent
  * @method Class Class based Component
  * @summary React component for rendering the card components
  * @param {object}  Replies contains the Replies of a comment
  * @param {Object} props - React PropTypes
  * @property {String|Object} className - String className compatible object for styling
  * @return {Node} React node containing comment form view
  */
export class ReplyCommentForm extends Component {
  state = {
    content: '',
    replyError: '',
  }

  onChange = (e) => {
    this.setState({
      [e.target.name]: e.target.value,
      replyError: '',
    });
  }

  onBlur = (e) => {
    if (e.target.value === '') {
      this.setState({
        replyError: 'Reply Content cannot be Empty',
      });
    } else {
      this.setState({
        formIsValid: true,
        replyError: '',
      });
    }
  }

  submitForm = (e) => {
    e.preventDefault();
    const { user } = this.props.auth;
    const { commentId } = this.props;
    const { formIsValid, content } = this.state;
    if (content.length >= 1500) {
      this.setState({
        formIsValid: false,
        replyError: "Can't have more than 1500 characters",
      });
    } else if (formIsValid) {
      const data = { content };
      this.props.postCommentReplies(commentId, data, user.image);
      this.setState({
        content: '',
      });
    } else {
      this.setState({
        replyError: 'Reply Content cannot be Empty',
      });
    }
  }

  /**
     * Renders the component.
     *
     * @memberof app.components.Replies
     * @return {string} - HTML markup for the component
   */
  render() {
    const { content, replyError } = this.state;
    return (
      <div>
        <div className='error_msg'>
          {' '}
          {replyError !== '' && <Alert type='warning' message={this.state.replyError} title='Invalid Credenntials' /> }
        </div>
        <div className='reply_input'>
          <form className='' onSubmit={this.submitForm}>
            <FormGroup controlId="formControlsTextarea">
              <FormControl
                onBlur={this.onBlur}
                onChange={this.onChange}
                componentClass="textarea"
                value={content}
                name='content'
                className='reply_form'
                rows="5"
                cols="80"
                placeholder="textarea"
              />
            </FormGroup>
            <Button type="reply" Class='btn__reply' label="Add Reply" />
          </form>
        </div>
      </div>
    );
  }
}

ReplyCommentForm.propTypes = {
  auth: PropTypes.object.isRequired,
};
/**
 * @method module:Reactator.ReduxContainerBuilderMapStateToProps
 *
 * @param {function} mapStateToProps - function mapping redux state to props
 *
 * @return {ReduxContainerBuilder} this builder
*/
const mapStateToProps = state => ({
  auth: state.auth,
});

export default connect(mapStateToProps, { postCommentReplies })(ReplyCommentForm);
