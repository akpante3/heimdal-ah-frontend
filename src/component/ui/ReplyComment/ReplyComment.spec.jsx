/* eslint-disable no-undef */
/* eslint-disable react/react-in-jsx-scope */
import { shallow } from 'enzyme';
import { ReplyComment } from './ReplyComment';
import { ReplyCommentCard } from './ReplyCommentCard/ReplyCommentCard';
import { ReplyCommentForm } from './ReplyCommentForm/ReplyCommentForm';

describe('rendering', () => {
  let wrapper;
  let props;
  beforeEach(() => {
    props = {
      getCommentReplies: jest.fn(),
      replies: '',
    };
  });

  it('renders One Comment Card components', () => {
    wrapper = shallow(<ReplyComment {...props} />);
    expect(wrapper.find(ReplyCommentCard)).toEqual({});
  });

  it('renders One Comment Form components', () => {
    wrapper = shallow(<ReplyComment {...props} />);
    expect(wrapper.find(ReplyCommentForm)).toEqual({});
  });

  it('should render the component', () => {
    wrapper = shallow(<ReplyComment {...props} />);
    expect(wrapper.length).toBe(1);
  });

  it('should render the component', () => {
    wrapper = shallow(<ReplyComment {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
