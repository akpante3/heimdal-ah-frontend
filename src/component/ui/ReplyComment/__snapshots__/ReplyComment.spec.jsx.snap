// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should render the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReplyComment
    getCommentReplies={[MockFunction]}
    replies=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="reply_button"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="searchbox"
          tabIndex="-1"
        >
          <FontAwesome
            className="fav_icons"
            name="comments"
          />
          Reply
        </div>,
        <div />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <FontAwesome
              className="fav_icons"
              name="comments"
            />,
            "Reply",
          ],
          "className": "reply_button",
          "onClick": [Function],
          "onKeyPress": [Function],
          "role": "searchbox",
          "tabIndex": "-1",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "fav_icons",
              "name": "comments",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "Reply",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="reply_button"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="searchbox"
            tabIndex="-1"
          >
            <FontAwesome
              className="fav_icons"
              name="comments"
            />
            Reply
          </div>,
          <div />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <FontAwesome
                className="fav_icons"
                name="comments"
              />,
              "Reply",
            ],
            "className": "reply_button",
            "onClick": [Function],
            "onKeyPress": [Function],
            "role": "searchbox",
            "tabIndex": "-1",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "fav_icons",
                "name": "comments",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "Reply",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
